src/runtime/rpc/network.cpp

```diff
@@ -75,6 +79,17 @@ void rpc_session::set_connected()
     on_rpc_session_connected.execute(this);
 }
 
+void rpc_session::set_negotiation()
+{
+    dassert(is_client(), "must be client session");
+
+    {
+        utils::auto_lock<utils::ex_lock_nr> l(_lock);
+        dassert(_connect_state == SS_CONNECTING, "session must be connecting");
+        _connect_state = SS_NEGOTIATION;
+    }
```

Comment:
```suggestion
    utils::auto_lock<utils::ex_lock_nr> l(_lock);
    dassert(_connect_state == SS_CONNECTING, "session must be connecting");
    _connect_state = SS_NEGOTIATION;
```

Reply:
I prefer to put these code insice `{}`， because someone else oneday will add some code in the end of `rpc_session::set_negotiation` and maybe will not notice the lock here. It will make this lock locks for a long time.